internal/component/unit.go

```diff
@@ -188,3 +216,125 @@ func (u *Unit) Dependents() Components {
 
 	return u.dependents
 }
+
+// String renders this unit as a human-readable string for debugging.
+func (u *Unit) String() string {
+	// Snapshot values under read lock to avoid data races
+	u.rLock()
+	path := u.path
+	deps := make([]string, 0, len(u.dependencies))
+
+	for _, dep := range u.dependencies {
+		deps = append(deps, dep.Path())
+	}
+
+	excluded := false
+	assumeApplied := false
+
+	if u.Execution != nil {
+		excluded = u.Execution.FlagExcluded
+		assumeApplied = u.Execution.AssumeAlreadyApplied
+	}
+
+	u.rUnlock()
+
+	return fmt.Sprintf(
+		"Unit %s (excluded: %v, assume applied: %v, dependencies: [%s])",
+		path, excluded, assumeApplied, strings.Join(deps, ", "),
+	)
+}
+
+// AbsolutePath returns the absolute path of the unit.
+// If path conversion fails, returns the original path and logs a warning if a logger is available.
+func (u *Unit) AbsolutePath() string {
+	if filepath.IsAbs(u.path) {
+		return u.path
+	}
```

Comment:
`filepath.Abs()` already contains code that looks like this:
```Go
	if IsAbs(path) {
		return Clean(path), nil
	}
```

if we do this check first then converted paths and non-converted paths will be different in whether they are `Clean()`'d. If this is desired then fine (e.g. "if already absolute then don't touch it"), but if there isn't a perceived downside I'd rather the behavior be as similar as possible between the two.


```suggestion
```

if we simply delete these three lines then we'll get the behavior I'm asking for.